### PR TITLE
roachtest: prevent configuring timeouts above 18h

### PR DESCRIPTION
**Pull request by tbg**
- build: drop timeout in roachtest nightly
- roachtest: reduce tpch timeout to 18h
- roachtest: enforce max timeout of 18h for non-weekly tests

### CODE REVIEW

**Comment by tbg** [comment]
cc @yuzefovich about reducing the timeout from 24h to 18h for this tpch test.

File: pkg/cmd/roachtest/test_registry.go [hunk 1]
```diff
@@ -127,6 +128,25 @@ func (r *testRegistryImpl) prepareSpec(spec *registry.TestSpec) error {
 	}
 	spec.Tags = append(spec.Tags, "owner-"+string(spec.Owner))
 
+	// At the time of writing, we expect the roachtest job to finish within 24h.
+	// Since each individual test may not be scheduled until a few hours in,
+	// individual tests should expect to take "less time". Longer-running
+	// tests require the weekly tag.
```
Comment (stevendanna) [inline pkg/cmd/roachtest/test_registry.go:134]: Is it worth mentioning that Teamcity has a 21 hour timeout (per the old comment)?

File: pkg/cmd/roachtest/test_registry.go [hunk 2]
```diff
@@ -127,6 +128,25 @@ func (r *testRegistryImpl) prepareSpec(spec *registry.TestSpec) error {
...
+	if spec.Timeout > maxTimeout {
+		var weekly bool
+		for _, tag := range spec.Tags {
+			if tag == "weekly" {
+				weekly = true
+			}
+		}
+		if !weekly {
+			return fmt.Errorf(
+				"timeout %s for %s exceeds the maximum allowed timeout of %s", spec.Timeout, spec.Name, maxTimeout,
```
Comment (stevendanna) [inline pkg/cmd/roachtest/test_registry.go:145]: The rest of the errors from this function are in the form of `("%s: timeout ..", spec.Name...)` if we wanted to stay consistent.

File: pkg/cmd/roachtest/tests/tpch_concurrency.go [hunk 3]
```diff
@@ -191,8 +191,8 @@ func registerTPCHConcurrency(r registry.Registry) {
 			// By default, the timeout is 10 hours which might not be sufficient
 			// given that a single iteration of checkConcurrency might take on
 			// the order of one hour, so in order to let each test run to
-			// complete we'll give it 24 hours.
-			Timeout: 24 * time.Hour,
+			// complete we'll give it 18 hours.
+			Timeout: 18 * time.Hour,
```
Comment (stevendanna) [inline pkg/cmd/roachtest/tests/tpch_concurrency.go:195]: For context, this test is currently taking ~6 hours when successful in GCE.

**Review by stevendanna — APPROVED** [review]
Seems reasonable to me.  

One worry might be whether this puts us at risk of hitting the TeamCity level timeout without getting a stacktrace. But I agree with the point you make in the first commit message -- we should probably teach roachtest how to enforce an overall timeout if we end up hitting the teamcity timeout.

For context, I took a peak at some of the recent builds. Currently, the nightly GCE tests take 12-16 hours to complete with the longest successful test (tpccbench) taking ~9 hours.

**Review by yuzefovich — COMMENTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72601)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @srosenberg, @stevendanna, and @tbg)

___
*[pkg/cmd/roachtest/tests/tpch_concurrency.go, line 195 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/72601#-Mo8RSFxNS4ZLWQzjmbu-r3-195:-Mo9Yvat4Fk4fdmn6N3o:b-nixqo8) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2a90ce6a657a9b67b40fa7ce60c6d167ea8dcce/pkg/cmd/roachtest/tests/tpch_concurrency.go#L195)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

For context, this test is currently taking ~6 hours when successful in GCE.
</blockquote></details>

Yeah, this change seems reasonable.

When we were working on introducing this test, in some versions we had more iterations for each query,  and we have seen some runs take on the order of 12 hours, so I just decided to double that for the timeout. Since then we reduced the number of iterations, so now it should be taking on the order of 6-8 hours usually.

**Review by srosenberg — APPROVED** [review]

**Comment by tbg** [comment]
note to self: I think I didn't push the edits yet, do that before merge.

**Comment by tbg** [comment]
Pushed the update,

bors r=[stevendanna,srosenberg]

**Pull request merged**